---
  title: "Leaf Oviposition 16S June2021"
author: "JReceveur"
output: 
  html_document:
    code_folding: hide
    toc: true
    toc_depth: 3
    toc_float: true
---
  
  ```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=14, fig.height=10)
knitr::opts_chunk$set(fig.align="center")
```

3,633,874 reads were obtained from 61 samples. 
Samples were rarefied to 10k reads
After filtering and rarefaction to 10k
#Import{.tabset}
##Sample Overview
``` {r import, message=FALSE, warning=FALSE,echo=TRUE}
set.seed(3246)
#Parse Silva function

parse_taxonomy_silva_138 <- function(char.vec){
  # Use default to assign names to elements in case problem with greengenes prefix
  char.vec = parse_taxonomy_default(char.vec)
  # Check for unassigned taxa
  if (char.vec["Rank1"] == "Unassigned") {
    char.vec <- c(Rank1="d__Unassigned", Rank2="d__Unassigned", Rank3="d__Unassigned", Rank4="d__Unassigned",
                  Rank5="d__Unassigned", Rank6="d__Unassigned", Rank7="d__Unassigned")
  }
  # Define the meaning of each prefix according to SILVA taxonomy
  Tranks = c(Rank1="Kingdom", Rank2="Phylum", Rank3="Class", Rank4="Order", Rank5="Family", Rank6="Genus", Rank7="Species")
  # Check for prefix using regexp, warn if there were none. trim indices, ti
  ti = grep("[[:alpha:]]\\_\\_", char.vec)
  if( length(ti) == 0L ){
    warning(
      "No silva prefixes were found. \n",
      "Consider using parse_taxonomy_delfault() instead if true for all OTUs. \n",
      "Dummy ranks may be included among taxonomic ranks now."
    )
    # Will want to return without further modifying char.vec
    taxvec = char.vec
    # Replace names of taxvec according to prefix, if any present...
  } else {
    # Format character vectors for Ambiguous taxa
    if( length(ti) < 7 ){
      for (key in names(char.vec)) {
        if ( char.vec[key] == "Ambiguous_taxa" ) {
          tax_no <- (as.numeric(substr(key, 5, 5)) - 1)
          char.vec[key] = sprintf("d__Ambiguous_taxa", tax_no)
        }
      }
      # Reset the trimmed indicies if Ambiguous taxa
      ti = grep("[[:alpha:]]\\_\\_", char.vec)
    }
    # Remove prefix using sub-"" regexp, call result taxvec
    taxvec = gsub("[[:alpha:]]\\_\\_", "", char.vec)
    # Define the ranks that will be replaced
    repranks = Tranks[substr(char.vec[ti], 1, 3)]
    # Replace, being sure to avoid prefixes notK present in Tranks
    names(taxvec)[ti[!is.na(repranks)]] = repranks[!is.na(repranks)]
  }
  return(taxvec)
}


#Load the packages needed
#To install these packages, run the command install.packages("package name here"), you can also install multiple packages at a time (see R documentation online for more info)
library(vegan)
library(MASS)
library(ggplot2)
library(plyr)
library(dplyr)
library(magrittr)
library(scales)
library(grid)
library(reshape2)
library(phyloseq)
library(randomForest)
library(knitr)
library(ape)
library(ggpubr)
library(viridis)
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#000000","#CC79A7") #Create a user defined color palette 
theme_set(theme_bw(base_size = 18)+theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()))#Set the general plotting theme see gggplot2 package for more info on other themes
biom=import_biom("LOLeafMar2021_10k.biom",parseFunction=parse_taxonomy_silva_138)
colnames(tax_table(biom))<-c(Rank1="Kingdom", Rank2="Phylum", Rank3="Class", Rank4="Order", Rank5="Family", Rank6="Genus", Rank7="Species")
#Import biom file and use the parse function built above
#If you put the file inside the folder of the R project, you can call it by just typing the file name as in the line above. IF its somewhere else and you need to find the path use the command file.choose() to bring up a browser to find the file path.
#biom
#tax_table(biom)
#tax_table(biom) <- tax_table(biom)[,-c(5:10,14)]#remove dummy ranks #Not needed

metadata=read.table("LeafOviMetadataMar2021.txt",header = TRUE) #Pull in your metadata and name the R object as metadata
metadata$Treatment = factor(metadata$Treatment, levels = c("OAK","SYC","HON","MIX","WAT","NC")) #fixes x-axis labels

tree=read_tree("LOTreeMar2021.nwk") #Read in the phylogenetic tree of the bacteria that was built in earlier processing


#This next few lines takes the individual files and combines them into a single object to make it easier to work with
sampdat=sample_data(metadata)
sample_names(sampdat)=metadata$id
physeq=merge_phyloseq(biom,sampdat,tree)

physeq<-filter_taxa(physeq, function (x) {sum(x > 0) > 1}, prune=TRUE)#Remove singletons and zeros
physeq
physeq<-subset_samples(physeq,Treatment!="WAT") #Removal of water only samples (rarefaction already removed extraction and sequencing controls, none above 1,400 reads)
physeq

#Currently physeq is the name of the object that combines your biom table (the number of sequences), the metadata and taxonomy
```

##Metadata
```{r}
metadata
```

#Alpha Diversity{.tabset}


##Observed Species
```{r}
ObservedSpecies<-plot_richness(physeq, x="Treatment",color="Treatment", measures=c("Observed"))+ylab("Observed Species")+geom_boxplot()+scale_fill_manual(cbPalette)+ scale_color_brewer(palette = "Dark2")
ObservedSpecies
ObservedRichness<-ObservedSpecies$data
head(ObservedRichness)

kruskal.test(value~Treatment, data=ObservedRichness)
ddply(ObservedRichness, c("Treatment"), summarise,
      N    = length(value),
      mean = mean(value),
      sd   = sd(value),
      se   = sd / sqrt(N)
)
ggplot(ObservedRichness, aes(x=Treatment,y= value))+geom_boxplot()

#plot richness is an already built function from the phyloseq package, to get more information about any package you can use the command ?packagename for example typing  (and running) ?phyloseq  will bring up the phyloseq help



```

##Inverse Simpson
```{r}
InvSimpson<-plot_richness(physeq, x="Treatment",color="Treatment", measures=c("invsimpson"))+ylab("Inverse Simpson ")+scale_fill_manual(cbPalette)+geom_boxplot()
InvSimpson
SimpsonData<-InvSimpson$data

kruskal.test(value~Treatment, data=SimpsonData)
#chi-sq=13.05,P = 0.0045
compare_means(value~Treatment, data=SimpsonData,method = "wilcox.test",p.adjust.method="fdr")


ddply(ObservedRichness, c("Treatment"), summarise,
      N    = length(value),
      mean = mean(value),
      sd   = sd(value),
      se   = sd / sqrt(N)
)
SimpsonPlot<-ggplot(SimpsonData, aes(x=Treatment,y= value,color=Treatment))+geom_boxplot()+ylab("Inverse Simpson (1/D)")+ scale_color_brewer(palette = "Dark2")+theme(legend.position = "none")+xlab("Leachate Type")

SimpsonPlot
```

##Shannon Diversity
```{r}
Shannon<-plot_richness(physeq, x="Treatment",color="Treatment", measures=c("shannon"))+ylab("Shannon Diversity")+scale_fill_manual(cbPalette)+geom_boxplot()
Shannon
ShannonData<-Shannon$data

kruskal.test(value~Treatment, data=ShannonData)
compare_means(value~Treatment, data=ShannonData,method = "wilcox.test",p.adjust.method="fdr")
#chi-sq = 10.99, P = 0.012


ShannonPlot<-ggplot(ShannonData, aes(x=Treatment,y= value,color=Treatment))+geom_boxplot()+ylab("Shannon Diversity")+xlab("Leachate Type")+theme(legend.position = "none")+scale_color_brewer(palette = "Dark2")
ShannonPlot



```



#Taxa Plots{.tabset}

*Note: Other taxa (present at an abundance lower than the cutoff, for example 0.1% at the phylum level) make up the remainder of the relative abundance for each sample (100%). While not shown here, will be combined into an other category for any actual manuscript figures.
```{r taxafiltering}


GPr  = transform_sample_counts(physeq, function(x) x / sum(x) ) #transform samples based on relative abundance
GPrPhylum=tax_glom(GPr, "Phylum")
PhylumLevel = filter_taxa(GPrPhylum, function(x) mean(x) > 1e-3, TRUE) #filter out any taxa lower tha 0.1%
GPrFamily=tax_glom(GPr,"Family")
FamilyLevel = filter_taxa(GPrFamily, function(x) mean(x) > 1e-2, TRUE) #filter out any taxa lower tha 1%
GPrGenus=tax_glom(GPr,"Genus")
GenusLevel = filter_taxa(GPrGenus, function(x) mean(x) > 2e-2, TRUE) #filter out any taxa lower tha 2%
```

##Total Abundances across all samples
Relative Abundances of  bacterial phyla



```{r TotalABUs,warning=F}
GPr  = transform_sample_counts(physeq, function(x) x / sum(x) ) #transform samples based on relative abundance
PhylumAll=tax_glom(GPr, "Phylum")
PhylumLevel = filter_taxa(PhylumAll, function(x) mean(x) > 1e-2, TRUE) #filter out any taxa lower tha 1%
FamilyAll=tax_glom(GPr,"Family")
FamilyLevel = filter_taxa(FamilyAll, function(x) mean(x) > 3e-2, TRUE) #filter out any taxa lower than 3%
GenusAll=tax_glom(GPr,"Genus")
GenusLevel = filter_taxa(GenusAll, function(x) mean(x) > 1e-2, TRUE) #filter out any taxa lower tha 1%

df <- psmelt(PhylumAll)
df$Abundance=df$Abundance*100
Trtdata <- ddply(df, c("Phylum"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance),
                 sd   = sd(Abundance),
                 se   = sd / sqrt(N)
)
#ddply from the ddplyr package is being used here to calculate the mean, standard deviation and standard error of your samples, see below for other examples using it to do the same thing but also between different sample groups
Trtdata
print("Family level abundances across all samples (> 1% of total relative abundances")
df <- psmelt(FamilyLevel)
df$Abundance=df$Abundance*100
Trtdata <- ddply(df, c("Family"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance),
                 sd   = sd(Abundance),
                 se   = sd / sqrt(N)
)
Trtdata


```


##Phylum Level





```{r TreatmentPlot5,warning=F}
df <- psmelt(PhylumLevel)
df$Abundance=df$Abundance*100

compare_means(Abundance~Trial, data=df,group.by="Phylum",method = "kruskal.test",p.adjust.method="fdr")
compare_means(Abundance~Trial, data=df,group.by="Phylum",method = "wilcox.test",p.adjust.method="fdr")

Trtdata <- ddply(df, c("Phylum","Treatment","Trial"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance),
                 sd=sd(Abundance),
                 se=sd/sqrt(N)
)
Trtdata

```


```{r}


PhylumTrial=ggplot(Trtdata, aes(x=Treatment,y=mean))+geom_bar(aes(fill = Phylum),colour="black", stat="identity")+xlab("Leachate")+ylab("Relative Abundance (%, > 1%)") + theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+scale_fill_viridis_d()+facet_wrap(~Trial)
#scale_x_discrete(labels=c("0 hrs", "24 hrs", "48 hrs","72 hrs"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))#+ scale_fill_manual(values=cbPalette)
PhylumTrial


PhylumWOther<-read.csv("PhylumLevel1PerWithOtherAdded.csv",header=T)
PhylumWOther$Treatment = factor(PhylumWOther$Treatment, levels = c("OAK","SYC","HON","MIX")) #fixes x-axis labels

PhylumPlot=ggplot(PhylumWOther, aes(x=Treatment,y=mean))+geom_bar(aes(fill = Phylum),colour="black", stat="identity")+ylab("Relative Abundance (%, > 1%)") + theme(axis.text.x = element_text(angle = 45, hjust = 1),axis.title.x=element_blank())+scale_fill_viridis_d()
#scale_x_discrete(labels=c("0 hrs", "24 hrs", "48 hrs","72 hrs"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))#+ scale_fill_manual(values=cbPalette)
PhylumPlot

df <- psmelt(PhylumAll)
df$Abundance=df$Abundance*100
Trtdata <- ddply(df, c("Phylum","Treatment"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance)
)
HON<-subset(Trtdata,Treatment=="HON")
HON<-subset(HON,mean!=0)
uniqueHON<-unique(HON$Phylum)
uniqueHON
 # [1] "Abditibacteriota"  "Acidobacteriota"   "Actinobacteriota"  "Armatimonadota"    "Bacteroidota"      "Basidiomycota"     "Bdellovibrionota" 
 # [8] "Firmicutes"        "Patescibacteria"   "Planctomycetota"   "Proteobacteria"    "Unassigned"        "Verrucomicrobiota"


OAK<-subset(Trtdata,Treatment=="OAK")
OAK<-subset(OAK,mean!=0)

uniqueOak<-unique(OAK$Phylum)
uniqueOak
 # [1] "Abditibacteriota"  "Acidobacteriota"   "Actinobacteriota"  "Armatimonadota"    "Bacteroidota"      "Basidiomycota"     "Bdellovibrionota" 
 # [8] "Firmicutes"        "MAST-12"           "Myxococcota"       "Planctomycetota"   "Proteobacteria"    "Unassigned"        "Verrucomicrobiota"

MIX<-subset(Trtdata,Treatment=="MIX")
MIX<-subset(MIX,mean!=0)

uniqueMIX<-unique(MIX$Phylum)
uniqueMIX

# [1] "Abditibacteriota"  "Acidobacteriota"   "Actinobacteriota"  "Armatimonadota"    "Bacteroidota"      "Basidiomycota"     "Bdellovibrionota" 
#  [8] "Firmicutes"        "Gemmatimonadota"   "MAST-12"           "Myxococcota"       "Planctomycetota"   "Proteobacteria"    "Unassigned"       
# [15] "Verrucomicrobiota"


SYC<-subset(Trtdata,Treatment=="SYC")
SYC<-subset(SYC,mean!=0)

uniqueSYC<-unique(SYC$Phylum)
uniqueSYC
# 
#  [1] "Abditibacteriota"  "Acidobacteriota"   "Actinobacteriota"  "Armatimonadota"    "Bacteroidota"      "Basidiomycota"     "Bdellovibrionota" 
#  [8] "Desulfobacterota"  "Firmicutes"        "Myxococcota"       "Proteobacteria"    "Unassigned"        "Verrucomicrobiota"

print("N Phyla unique to Oak (compared to honeysuckle)")

length(subset(uniqueOak,!(uniqueOak %in% uniqueHON)))
subset(uniqueOak,!(uniqueOak %in% uniqueHON))

print("N Phyla unique to Oak (compared to SYC)")

length(subset(uniqueOak,!(uniqueOak %in% uniqueSYC)))
subset(uniqueOak,!(uniqueOak %in% uniqueSYC))

print("N Phyla unique to Oak (compared to MIX)")

length(subset(uniqueOak,!(uniqueOak %in% uniqueMIX)))
subset(uniqueOak,!(uniqueOak %in% uniqueMIX))

```





##Family Level




```{r}
df <- psmelt(FamilyLevel)
df$Abundance=df$Abundance*100


compare_means(Abundance~Treatment, data=df,group.by="Family",method = "kruskal.test",p.adjust.method="fdr")
compare_means(Abundance~Treatment, data=df,group.by="Family",method = "wilcox.test",p.adjust.method="fdr")


summary(aov(Abundance~Treatment,group.by="Family",data=df))
#Sig families Erwiniaceae, Oxalobacteraceae,
Means<-compare_means(Abundance~Treatment, data=df,group.by="Family",method = "wilcox.test",p.adjust.method="fdr")

Trtdata <- ddply(df, c("Family","Treatment"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance),
                 sd=sd(Abundance),
                 se=sd / sqrt(N)
)
Trtdata

FamilyPlot=ggplot(Trtdata, aes(x=Treatment,y=mean))+geom_bar(aes(fill = Treatment),colour="black", stat="identity")+
  xlab("Leachate")+ylab("Relative Abundance (%, > 1%)") + theme(axis.text.x = element_text(angle = 45, hjust = 1),legend.position="none",axis.title.x=element_blank())+
scale_fill_brewer(palette="Dark2")+facet_wrap(~Family)+geom_errorbar(aes(ymin=mean-se,ymax=mean+se))
#scale_x_discrete(labels=c("0 hrs", "24 hrs", "48 hrs","72 hrs"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))#+ scale_fill_manual(values=cbPalette)
FamilyPlot

#Multicomp letters for all families



FamilyPlot2=ggplot(Trtdata, aes(x=Treatment,y=mean))+geom_bar(aes(fill = Family),colour="black", stat="identity")+ylab("Relative Abundance (%, > 1%)") + theme(axis.text.x = element_text(angle = 45, hjust = 1),axis.title.x=element_blank())+
scale_fill_viridis_d()
FamilyPlot2


for (i in (Means$Family)){
  Tax<-i
  TaxAbundance<-subset(Means,Family==i )
  Hyphenated<-as.character(paste0(TaxAbundance$group1,"-",TaxAbundance$group2))
  difference<-TaxAbundance$p.adj
  names(difference)<-Hyphenated
  Letters<-multcompLetters(difference)
  #print(Letters)
  SigList[i]<-Letters

}
SigList
vec<-unlist(SigList)
vec<-vec[-1]

```


##Genus level







#Ordinations

-Ellipses with 95% CI for the mean of each group not shown due to sample size

##PCoA {.tabset}

###Weighted UniFrac Distance

```{r PCoA, warning=FALSE}
set.seed(1245)

ord=ordinate(physeq,"PCoA", "wunifrac")
ordplot=plot_ordination(physeq, ord,"samples", color="Treatment",shape="Treatment")+geom_point(size=5)+ scale_color_brewer(palette = "Dark2")+facet_wrap(~Trial)#+scale_colour_manual(values=cbPalette)+scale_fill_manual(values=cbPalette)
ordplot

set.seed(1245)
GPdist=phyloseq::distance(physeq, "wunifrac")

adonis(GPdist ~ Treatment*Trial, as(sample_data(physeq), "data.frame"))

set.seed(1245)
GPdist=phyloseq::distance(physeq, "wunifrac")

adonis(GPdist ~ Treatment, as(sample_data(physeq), "data.frame"))



Trial1<-subset_samples(physeq,Trial==1)
Trial2<-subset_samples(physeq,Trial==2)
Trial3<-subset_samples(physeq,Trial==3)
Trial4<-subset_samples(physeq,Trial==4)



GPdist=phyloseq::distance(Trial1, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(Trial1), "data.frame"))


GPdist=phyloseq::distance(Trial2, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(Trial2), "data.frame"))


GPdist=phyloseq::distance(Trial3, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(Trial3), "data.frame"))


GPdist=phyloseq::distance(Trial4, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(Trial4), "data.frame"))

#OAK, HON,SYC,MIX
OAKvHON <- subset_samples(physeq, Treatment %in% c("OAK", "HON"))
OAKvSYC <- subset_samples(physeq, Treatment %in% c("OAK", "SYC"))
OAKvMIX <- subset_samples(physeq, Treatment %in% c("OAK", "MIX"))

HONvSYC <- subset_samples(physeq, Treatment %in% c("HON", "SYC"))
HONvMIX <- subset_samples(physeq, Treatment %in% c("HON", "MIX"))

SYCvMIX <- subset_samples(physeq, Treatment %in% c("SYC", "MIX"))

.05/6

set.seed(1245)
GPdist=phyloseq::distance(OAKvHON, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(OAKvHON), "data.frame"))
#F1,22=7.02,P = .003

set.seed(1245)
GPdist=phyloseq::distance(OAKvSYC, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(OAKvSYC), "data.frame"))
#F1,22=5.05,P=.003

set.seed(1245)
GPdist=phyloseq::distance(OAKvMIX, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(OAKvMIX), "data.frame"))
#F=2.14,P = .118

set.seed(1245)
GPdist=phyloseq::distance(HONvSYC, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(HONvSYC), "data.frame"))
#F=2.87,P=.052


set.seed(1245)
GPdist=phyloseq::distance(HONvMIX, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(HONvMIX), "data.frame"))
#F=2.39,P=.11


set.seed(1245)
GPdist=phyloseq::distance(SYCvMIX, "wunifrac")
adonis(GPdist ~ Treatment, as(sample_data(SYCvMIX), "data.frame"))
#F=1.84,P =.15

```





#Beta diversity{.tabset}

##Multivariate Dispersions
Homogenieity of Multivariate Dispersions:
  
  ###Weighted UniFrac
  
```{r, warning=FALSE}
set.seed(1245)
#Overall test significant 
library("RColorBrewer")
GPdist=phyloseq::distance(physeq, "wunifrac")
beta=betadisper(GPdist, sample_data(physeq)$Treatment)
permutest(beta)
boxplot(beta,xlab="Leachate Type",col=brewer.pal(n = 4, name = "Dark2"))


dev.off()
tiff("Figures/BetadisprPlot.tiff", width = 100, height = 100, units = 'mm', res = 600)
boxplot(beta,xlab="Leachate Type",col=brewer.pal(n = 4, name = "Dark2"))
dev.off()

0.05/6 #Alpha value B-correction = 0.0083
set.seed(1245)

GPdist=phyloseq::distance(OAKvHON, "wunifrac")
beta=betadisper(GPdist, sample_data(OAKvHON)$Treatment)
permutest(beta)#F1,22=14.3,P=0.002
boxplot(beta)

set.seed(1245)

GPdist=phyloseq::distance(OAKvMIX, "wunifrac")
beta=betadisper(GPdist, sample_data(OAKvMIX)$Treatment)
permutest(beta)#F1,22=.73,P=.39
boxplot(beta)

set.seed(1245)

GPdist=phyloseq::distance(OAKvSYC, "wunifrac")
beta=betadisper(GPdist, sample_data(OAKvSYC)$Treatment)
permutest(beta)#f1,22=.69,P=.41
boxplot(beta,xlab="Leachate")

betadis



set.seed(1245)

GPdist=phyloseq::distance(HONvSYC, "wunifrac")
beta=betadisper(GPdist, sample_data(HONvSYC)$Treatment)
permutest(beta)#f1,22=10.67,P=.002
boxplot(beta)

set.seed(1245)

GPdist=phyloseq::distance(HONvMIX, "wunifrac")
beta=betadisper(GPdist, sample_data(HONvMIX)$Treatment)
permutest(beta)#F1,22=3.3,P=.1
boxplot(beta)

set.seed(1245)

GPdist=phyloseq::distance(SYCvMIX, "wunifrac")
beta=betadisper(GPdist, sample_data(SYCvMIX)$Treatment)
permutest(beta) #F1,22=.24,P=.64
boxplot(beta)



```

Split by Trial
```{r}
set.seed(1245)
GPdist=phyloseq::distance(physeq, "wunifrac")
beta=betadisper(GPdist, sample_data(physeq)$Trial)
permutest(beta)
boxplot(beta)


```



###Jaccard 

```{r, warning=FALSE}
set.seed(1245)

GPdist=phyloseq::distance(physeq, "jaccard")
beta=betadisper(GPdist, sample_data(physeq)$Species)
permutest(beta)
boxplot(beta)
```

Split by Species (P.regina)
```{r}
set.seed(1245)
physeqP.regina<-subset_samples(physeq,Species=="P.Regina")
GPdist=phyloseq::distance(physeqP.regina, "jaccard")
beta=betadisper(GPdist, sample_data(physeqP.regina)$Type)
permutest(beta)
boxplot(beta)


```

Split by Species (L.sericata)
```{r}
set.seed(1245)
physeqL.sericata<-subset_samples(physeq,Species=="L.Sericata")
GPdist=phyloseq::distance(physeqL.sericata, "jaccard")
beta=betadisper(GPdist, sample_data(physeqL.sericata)$Type)
permutest(beta)
boxplot(beta)


```


##PERMANOVA

###Weighted UniFrac

```{r,warning=F}
set.seed(1245)

GPdist=phyloseq::distance(physeq, "wunifrac")
adonis(GPdist ~ Species+Month+Type, as(sample_data(physeq), "data.frame"))
```

###Jaccard Distance

```{r,warning=F}
GPdist=phyloseq::distance(physeq, "jaccard")
adonis(GPdist ~ Species+Month+Type, as(sample_data(physeq), "data.frame"))
```

#Random Forest predictions{.tabset}



##Leachate type

```{r}
set.seed(1245)
ForestData=GenusLevel#Change this one so you dont have to rewrite all variables
predictors=t(otu_table(ForestData))
response <- as.factor(sample_data(ForestData)$Treatment)
rf.data <- data.frame(response, predictors)
MozzieForest <- randomForest(response~., data = rf.data, ntree = 1000,importance=T)

print(MozzieForest)#returns overall Random Forest results
imp <- importance(MozzieForest)#all the steps that are imp or imp. are building a dataframe that contains info about the taxa used by the Random Forest testto classify treatment 
imp <- data.frame(predictors = rownames(imp), imp)
imp.sort <- arrange(imp, desc(MeanDecreaseAccuracy))
imp.sort$predictors <- factor(imp.sort$predictors, levels = imp.sort$predictors)
imp.20 <- imp.sort[1:20, ]#


destroyX = function(es) {
  f = es
  for (row in c(1:nrow(f))){ #for each column in dataframe
    if (startsWith(row.names(f)[row], "X") == TRUE)  { #if starts with 'X' ..
      row.names(f)[row] <- substr(row.names(f)[row], 2, 100) #get rid of it
    }
    assign(deparse(substitute(es)), f, inherits = TRUE)
  }
}
destroyX(imp.20)


```



###Top 10 genus level indicator taxa for differences between Urban V Rural (MDA)

```{r}
imp.10 <- imp.sort[1:10, ]#
destroyX(imp.10)
otunames <- row.names(imp.10)
r <- rownames(tax_table(ForestData)) %in% otunames
GenusRandomForestSubset = subset_taxa(GenusAll, row.names(tax_table(GenusAll))%in% otunames)
#GenusRandomForestSubset

df <- psmelt(GenusRandomForestSubset)
df$Abundance=df$Abundance*100
compare_means(Abundance ~ Treatment, data = df, group.by = "Genus", p.adjust.method = "fdr",method="kruskal.test")
Means<-compare_means(Abundance ~ Treatment, data = df, group.by = "Genus", p.adjust.method = "fdr",method="wilcox.test")


SigList<-list()
for (i in (Means$Genus)){
  Tax<-i
  TaxAbundance<-subset(Means,Genus==i )
  Hyphenated<-as.character(paste0(TaxAbundance$group1,"-",TaxAbundance$group2))
  difference<-TaxAbundance$p.adj
  names(difference)<-Hyphenated
  Letters<-multcompLetters(difference)
  #print(Letters)
  SigList[i]<-Letters

}
SigList
vec<-unlist(SigList)
vec<-vec[-1]








Trtdata <- ddply(df, c("Genus","Treatment"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance),
                 sd=sd(Abundance),
                 se=sd/sqrt(N)
)
Trtdata
Trtdata[1,1]<- "Allorhizobium"
Trtdata[2,1]<- "Allorhizobium"
Trtdata[3,1]<- "Allorhizobium"
Trtdata[4,1]<- "Allorhizobium"


Trtdata[13,1]<-"Clostridum ss1"
Trtdata[14,1]<-"Clostridum ss1"
Trtdata[15,1]<-"Clostridum ss1"
Trtdata[16,1]<-"Clostridum ss1"

Trtdata

cdataplot=ggplot(Trtdata, aes(x=Treatment,y=mean))+geom_bar(aes(fill = Treatment),colour="black", stat="identity")+ facet_wrap(~Genus)+xlab("Leachate")+ylab("Relative Abundance (%)") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+scale_fill_brewer(palette = "Dark2")+geom_errorbar(aes(ymin=mean-se,ymax=mean+se))+
  theme(legend.position = "none",axis.title.x=element_blank())
cdataplot


```

##Trial

```{r}
#GenusAll
set.seed(1245)
ForestData=GenusAll#Change this one so you dont have to rewrite all variables
predictors=t(otu_table(ForestData))
response <- as.factor(sample_data(ForestData)$Trial)
rf.data <- data.frame(response, predictors)
MozzieForest <- randomForest(response~., data = rf.data, ntree = 1000,importance=T)
print(MozzieForest)#returns overall Random Forest results
imp <- importance(MozzieForest)#all the steps that are imp or imp. are building a dataframe that contains info about the taxa used by the Random Forest testto classify treatment 
imp <- data.frame(predictors = rownames(imp), imp)
imp.sort <- arrange(imp, desc(MeanDecreaseAccuracy))
imp.sort$predictors <- factor(imp.sort$predictors, levels = imp.sort$predictors)
imp.10 <- imp.sort[1:10, ]#

destroyX(imp.10)


otunames <- row.names(imp.10)
r <- rownames(tax_table(ForestData)) %in% otunames
df <- psmelt(GenusAll)
#head(df)
GenusRandomForestSubset = subset_taxa(GenusAll, row.names(tax_table(GenusAll))%in% otunames)
GenusRandomForestSubset

df <- psmelt(GenusRandomForestSubset)
df$Abundance=df$Abundance*100

Trtdata <- ddply(df, c("Genus","Type","sample_Species","Month"), summarise,
                 N    = length(Abundance),
                 mean = mean(Abundance)
)

compare_means(Abundance ~ sample_Species, data = df, group.by = "Genus", p.adjust.method = "fdr",method="kruskal.test")


cdataplot=ggplot(Trtdata, aes(x=Month,y=mean))+geom_bar(aes(fill = Genus),colour="black", stat="identity")+ facet_grid(sample_Species~Type)+xlab("Time")+ylab("Relative Abundance (%)") + theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+scale_fill_viridis_d()
cdataplot

```
